lib/btree.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+// And [degree - 1 ... 2 * degree - 1] + 1 "empty" elements
+
+class Element {
+  constructor(value, tuple, less = null) {
+    this.value = value;
+    this.tuple = tuple;
+    this.less = less;
+  }
+}
+
+const empty = less => new Element(undefined, null, less);
+
+const destroyNode = (parent, index) => {
+  const node = parent[index].less;
+  const len = node.length;
+  const newLeftNode = node.splice(0, len / 2 - 1); // First half
+  const mid = node.shift();
+  newLeftNode.push(empty(mid.less));
+  mid.less = newLeftNode;
+  parent.splice(index, 0, mid);
+  return parent;
+};
+
+const isLeaf = node => !node[0].less;
+
+const binarySearch = (node, value) => {
+  let start = 0;
+  let end = node.length - 1;
+  while (start <= end) {
+    const i = (start + end) >> 1;
+    const item = node[i];
+    const itemValue = item.value;
+    if (value > itemValue) start = i + 1;
+    else if (!(value >= itemValue)) end = i - 1;
+    // undefined must be bigger then other value
+    // (!(1 >= undefined)) -> true, but just (1 < undefined) -> false
+    else return [true, item.tuple];
+  }
+  return [false, start];
+};
+
+function* inorderTraversal(start, finish, currNode) {
+  let [found, startIndex] = binarySearch(currNode, start);
+  if (found) startIndex++;
+  let [_, finishIndex] = binarySearch(currNode, finish);
+  if (finish === undefined)
+    finishIndex = currNode.length - 1;
+  for (let currIndex = startIndex; currIndex <= finishIndex; currIndex++) {
+    const currElement = currNode[currIndex];
+    if (!(isLeaf(currNode)))
+      yield* inorderTraversal(start, finish, currElement.less);
+    if (currIndex < finishIndex)
+      yield currNode[currIndex].tuple;
+  }
+
+}
+
+class BTree {
+
+  constructor(degree = DEFAULT_DEGREE) {
+    this.root = [empty()];
+    this.minDegree = degree;
+  }
+
+  get(value) {
+    let currNode = this.root;
+    while (currNode) {
+      const [found, i] = binarySearch(currNode, value);
+      if (found)
+        return i;
+      currNode = currNode[i].less;
+    }
+    return null;
+  }
+
+  insert(value, tuple) {
+    const alreadyHaveErrMes = `value ${value} already is in the tree`;
```

Comment:
Move this to the place it is used in, you don't have to create it always. It's better to copy the error here (or make a function that will return this error) that create it every time we enter this function.